samanthas_telegram_bot/bot_data_loader/bot_data_loader.py

```diff
@@ -0,0 +1,236 @@
+        in user's language).
+
+        These words have to be assigned to these age ranges for bot to display the correct phrase.
+        For example: `phrases.csv` contains the phrase with ID ``option_adults``, so the age range
+        corresponding to adults has to be assigned ``bot_phrase_id: option_adults``.
+        """
+
+        data = cls._get_json(url_infix="age_ranges")
+
+        age_ranges: dict[AgeRangeType, tuple[AgeRange, ...]] = {
+            type_: tuple(AgeRange(**item) for item in data if item["type"] == type_)
+            for type_ in (AgeRangeType.STUDENT, AgeRangeType.TEACHER)
+        }
+
+        # add IDs of bot phrases to teachers' age ranges
+        age_to_phrase_id = {
+            5: "young_children",
+            9: "older_children",
+            13: "adolescents",
+            18: "adults",
+            66: "seniors",
+        }
+        for age_range in age_ranges[AgeRangeType.TEACHER]:
+            age_range.bot_phrase_id = f"option_{age_to_phrase_id[age_range.age_from]}"
+
+        return age_ranges
+
+    @classmethod
+    def _get_assessments(cls, lang_code: str) -> dict[int, Assessment]:
+        """Gets assessment questions from the backend, based on language.
+
+        Returns a dictionary matching an age range ID to assessment.
+        """
+
+        data = cls._get_json(
+            url_infix=API_URL_INFIX_ENROLLMENT_TESTS,
+            name_for_logger=f"assessments for {lang_code=}",
+            params={"language": lang_code},
+        )
+
+        assessments = tuple(
+            Assessment(
+                id=item["id"],
+                age_range_ids=tuple(item["age_ranges"]),
+                questions=tuple(
+                    AssessmentQuestion(
+                        id=question["id"],
+                        text=question["text"],
+                        options=tuple(
+                            AssessmentQuestionOption(**option) for option in question["options"]
+                        ),
+                    )
+                    for question in item["questions"]
+                ),
+            )
+            for item in data
+        )
+
+        # Each assessment has a sequence of age range IDs. We must match every single age range ID
+        # in this sequence to a respective assessment.
+        assessment_for_age_range_id = {
+            age_range_id: assessment
+            for assessment in assessments
+            for age_range_id in assessment.age_range_ids
+        }
+
+        return assessment_for_age_range_id
+
+    @classmethod
+    def _get_day_and_time_slots(cls) -> tuple[DayAndTimeSlot, ...]:
+        """Gets day and time slots from the backend."""
+
+        def get_hour(str_: str) -> int:
+            """Takes a string like 05:00:00 and returns hours (5 in this example)."""
+            return int(str_.split(":")[0])
+
+        data = cls._get_json(url_infix=API_URL_INFIX_DAY_AND_TIME_SLOTS)
```

Comment:
This should be abstracted away. BotData should not know anything about the specifics of the backend. An api call should be packed into a call to a specific api-calling class

Reply:
I'll think about it to avoid major refactoring right now but at the same time make it better.

Reply:
Done some refactoring that could be done without major overhaul